homeassistant/components/synology_dsm/camera.py

```diff
@@ -31,23 +31,23 @@
 
 
 async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable
 ) -> None:
     """Set up the Synology NAS cameras."""
 
     data = hass.data[DOMAIN][entry.unique_id]
-    api = data[SYNO_API]
+    api: SynoApi = data[SYNO_API]
 
     if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
         return
 
     # initial data fetch
-    coordinator = data[COORDINATOR_CAMERAS]
+    coordinator: DataUpdateCoordinator = data[COORDINATOR_CAMERAS]
     await coordinator.async_refresh()
```

Comment:
Add this check after `coordinator.async_refresh()`. Then `# type: ignore` on L50 can be removed.
```python
if coordinator.data is None:
        raise ConfigEntryNotReady
```

Reply:
Actually it's better to use `async_config_entry_first_refresh` here. And then `assert coordinator.data is not None` should be safe.